src/NuGet.Core/NuGet.Configuration/Settings/ImmutableSettings.cs

```diff
@@ -0,0 +1,62 @@
+// All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace NuGet.Configuration
+{
+    // Represents a wrapper for an immutable settings instance.
+    // This means that any methods invoked on this instance that try to alter it will throw.
+    internal class ImmutableSettings : ISettings
```

Comment:
Would it make sense to add an `IReadOnlySettings` that has only the methods that do not mutate settings? `ISettings` could extend `IReadOnlySettings` with the mutable methods.